src/Context/cartContext.jsx

```diff
@@ -0,0 +1,133 @@
+import { createContext, useContext, useReducer } from "react";
+import { cartReducer } from "../Reducer/cartReducer";
+import {useAuth} from "../Context/authContext";
+import { useEffect } from "react";
+import axios from "axios";
+
+
+
+const CartContext = createContext();
+
+const CartProvider = ({children}) => {
+
+   
+   
+    const [cartState, cartDispatch] = useReducer(cartReducer, {
+        cart: []
+    })
+    const {token, user} = useAuth();
+
+    useEffect(() => {
+        {
+            token && 
+            (async() => {
```

Comment:
Great work. Your code gave better understanding of POST request on which I'm working right now. Very well written code and good to merge.